### ISSUE TITLE
Linux arm64

### ISSUE THREAD

**Issue by metaskills** [asg017/sqlite-vss#82]
Using a M2 Mac with development containers. Target deploy will be an AWS Lambda which can be x86 or arm64. However, I'm kind of blocked right now developing since all my devcontainers just naturally match my current platform, arm64. Looking into docs now on compiling, will report any issues... https://github.com/asg017/sqlite-vss/blob/main/docs.md

**Comment by metaskills** [comment]
Trying to debug this.

```
> make loadable-release
make: *** No rule to make target `loadable-release'.  Stop.
```

**Comment by asg017** [comment]
Can you `ls` to make sure there's a Makefile in your directory? Could also try `make loadable` but both should work

**Comment by metaskills** [comment]
Some updates on this. First, the "No rule to make target" was my own fault. I forgot to cd up a few directories. I've been having a hard time with Amazon Linux 2 (I always do) WRT getting things built from source. Things were looking good like they may work in my devcontainer with this:

```dockerfile
FROM public.ecr.aws/sam/build-nodejs18.x
RUN yum update -y && \
    yum install -y libgomp atlas-devel lapack-devel sqlite-devel cmake3 && \
    /usr/sbin/update-alternatives --install /usr/bin/cmake cmake /usr/bin/cmake3 1
```

Like I had to get cmake 3 installed vs the default of v2. For record keeping, here is where things ended up.

* libgomp-7.3.1-15.amzn2.aarch64
* atlas-devel-3.10.1-12.amzn2.0.2.aarch64
* lapack-devel-3.4.2-8.amzn2.0.2.aarch64
* sqlite-devel.aarch64 0:3.7.17-8.amzn2.1.2
* cmake3.aarch64 0:3.13.1-1.amzn2

However, I kept hitting this with `make loadable-release`. I even tried installing gcc v9 instead of AML2 v7. No luck.

```
mkdir -p dist/debug
mkdir -p dist/release
cmake -DCMAKE_BUILD_TYPE=Release -B build_release; make -C build_release
CMake Error: The source directory "" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
make: *** build_release: No such file or directory.  Stop.
make: *** [dist/release/vector0.so] Error 2
```

I'm gonna pivot a bit and fork the project and see if I can contribute a pull request for building with GitHub Actions and QEMU. I've had tons of experience doing this in the https://github.com/rails-lambda projects for multi-platform support. It is a little slow but gets the job done till GitHub finally releases their own arm64 runners. 

Any thoughts?

**Comment by metaskills** [comment]
Currently testing building on `ghcr.io/catthehacker/ubuntu:act-22.04` which mimics GitHub Actions runners and has the version of cmake needed as well as arm64 image support. Found this along the way:

- https://github.com/asg017/sqlite-vss/issues/14

**Comment by metaskills** [comment]
OK, here is where I am. Summary with details below:

* I have created a `sqlite-vss-linux-arm64` test package, but does not work on AL2. Pivoting to Ubuntu.
* Are npm packages "static" libraries. Tests with `ldd` show dynamic links. Which is likely fine.

I created a fork here (https://github.com/metaskills/sqlite-vss/tree/LinuxArm64) with a single commit to prototype building a `sqlite-vss-linux-arm64` npm test package. I used the `ghcr.io/catthehacker/ubuntu:act-22.04` container from the act project (https://github.com/nektos/act). I chose this for a few reasons. 1) Mimics GitHub Actions Runners 2) Is a multi-platform image 3) CMake is of sufficient version. My full notes on the build and test package here:

https://github.com/metaskills/sqlite-vss/commit/474e2daf266792842866a47a9b9586f5c9129915#diff-fbc3aef752b70d1a9396dd71d18baf5e8a99b18ac31f8e8ba8143020bfe3be17

Will report back after I switch my Lambda Container project from AL2 to Ubuntu.

**Comment by metaskills** [comment]
SUCCESS! Had to use a Debian bookwork image to get GLIBC up far enough. Details here:

https://github.com/metaskills/sqlite-vss/commit/f2c95cdecd1d4e4ef9390ba6232b08c038063f7b

So, is GLIBC v2.32 or higher a requirement? Cloud not find any mention of it.

**Comment by neilkinnish** [comment]
@metaskills did you manage to deploy this on AWS lambda?

**Comment by metaskills** [comment]
No, not gotten that far. Should be easy, just package it up in a container using the same local version I built. Will get there in the coming days/weeks.

**Comment by asg017** [comment]
Hey @metaskills , thanks again for your hard work here! I do intend to dig into this + ship `linux-arm64` version of `sqlite-vss` to Python/Ruby/Node.js etc. I got sick this week so I'll be out for a few days, but fully intend to get back to this when I get better!

**Comment by metaskills** [comment]
Thanks, take your time. I am unblocked, so no rush. I still have a TON of AI learning to do before having a turn-key linux arm npm package would be a concern. Cheers!

**Comment by neilkinnish** [comment]
thanks @metaskills.

Had a play this morning with your fork... have it running on lambda 👍 

Hope you feel better @asg017

**Comment by adsharma** [comment]
Linux container on a arm64 windows laptop. I guess people using Apple containers to run Linux will face a similar issue:

```
uv pip install sqlite-vss
  × No solution found when resolving dependencies:
  ╰─▶ Because all versions of sqlite-vss have no wheels with a matching platform tag (e.g., `manylinux_2_39_aarch64`)
      and you require sqlite-vss, we can conclude that your requirements are unsatisfiable.

      hint: Pre-releases are available for `sqlite-vss` in the requested range (e.g., 0.1.2a3), but pre-releases
      weren't enabled (try: `--prerelease=allow`)

      hint: Wheels are available for `sqlite-vss` (v0.1.2) on the following platforms: `manylinux_2_17_x86_64`,
      `manylinux1_x86_64`, `manylinux2014_x86_64`, `macosx_10_6_x86_64`, `macosx_11_0_arm64`
```
